components/pagination/Pagination.tsx

```diff
@@ -0,0 +1,48 @@
+import Link from "next/link";
+import React from "react";
+import usePagination from "./usePagination";
+
+export type PaginationProps = {
+  totalItems: number
+  currentPage: number
+  renderPageLink: (page: number) => string
+  itemsPerPage?: number
```

Comment:
Maybe don't make this an optional prop? There isn't much of an interface for this component, and `itemsPerPage` is a core configuration of this component in general.